packages/core/src/external/saml/__tests__/saml-envelope.test.ts

```diff
@@ -0,0 +1,185 @@
+      bodyData: iti38BodyData,
+      publicCert: IHE_STAGING_CERT,
+    });
+
+    const signedEnvelope = signEnvelope({ xml: soapEnvelope, privateKey: IHE_STAGING_KEY });
+
+    const modifiedAssertion = signedEnvelope.replace(
+      /<saml2:Issuer Format="urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress">.*<\/saml2:Issuer>/,
+      '<saml2:Issuer Format="urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress">invalid-issuer</saml2:Issuer>'
+    );
+
+    expect(verifySaml({ xmlString: modifiedAssertion, publicCert: IHE_STAGING_CERT })).toBeFalsy();
+  });
+  it("should fail verification if the ITI38 SAML assertion signature is modified", () => {
+    const soapEnvelope = createITI38SoapEnvelope({
+      bodyData: iti38BodyData,
+      publicCert: IHE_STAGING_CERT,
+    });
+
+    const signedEnvelope = signEnvelope({ xml: soapEnvelope, privateKey: IHE_STAGING_KEY });
+
+    const modifiedSignature = signedEnvelope.replace(
+      /<ds:SignatureValue>.*<\/ds:SignatureValue>/,
+      "<ds:SignatureValue>invalid-signature</ds:SignatureValue>"
+    );
+
+    expect(verifySaml({ xmlString: modifiedSignature, publicCert: IHE_STAGING_CERT })).toBeFalsy();
+  });
+  it("should fail verification if the ITI38 SAML assertion digest is modified", () => {
+    const soapEnvelope = createITI38SoapEnvelope({
+      bodyData: iti38BodyData,
+      publicCert: IHE_STAGING_CERT,
+    });
+
+    const signedEnvelope = signEnvelope({ xml: soapEnvelope, privateKey: IHE_STAGING_KEY });
+
+    const modifiedDigest = signedEnvelope.replace(
+      /<ds:DigestValue>.*<\/ds:DigestValue>/,
+      "<ds:DigestValue>invalid-digest</ds:DigestValue>"
+    );
+
+    expect(verifySaml({ xmlString: modifiedDigest, publicCert: IHE_STAGING_CERT })).toBeFalsy();
+  });
+  it("should sign and verify the ITI39 envelope successfully with envelope", () => {
+    const soapEnvelope = createITI39SoapEnvelope({
+      bodyData: iti39BodyData,
+      publicCert: IHE_STAGING_CERT,
+    });
+
+    const signedEnvelope = signEnvelope({ xml: soapEnvelope, privateKey: IHE_STAGING_KEY });
+
+    expect(verifySaml({ xmlString: signedEnvelope, publicCert: IHE_STAGING_CERT })).toBeTruthy();
+  });
+  it("should fail verification if the ITI39 SAML assertion is modified", () => {
+    const soapEnvelope = createITI39SoapEnvelope({
+      bodyData: iti39BodyData,
+      publicCert: IHE_STAGING_CERT,
+    });
+
+    const signedEnvelope = signEnvelope({ xml: soapEnvelope, privateKey: IHE_STAGING_KEY });
+
+    const modifiedAssertion = signedEnvelope.replace(
+      /<saml2:Issuer Format="urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress">.*<\/saml2:Issuer>/,
+      '<saml2:Issuer Format="urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress">invalid-issuer</saml2:Issuer>'
+    );
+
+    expect(verifySaml({ xmlString: modifiedAssertion, publicCert: IHE_STAGING_CERT })).toBeFalsy();
+  });
+  it("should fail verification if the ITI39 SAML assertion signature is modified", () => {
```

Comment:
nit: we add empty spaces within the function, so one would expect that we'd have spaces within functions too, as they are of a higher level :)

Reply:
I just removed all the excessive empty spaces